llvm/tools/sycl-post-link/CompileTimePropertiesPass.cpp

```diff
@@ -254,33 +256,55 @@ PreservedAnalyses CompileTimePropertiesPass::run(Module &M,
     }
   }
 
-  // Process all properties on kernels.
   for (Function &F : M) {
     // Only consider kernels.
     if (F.getCallingConv() != CallingConv::SPIR_KERNEL)
       continue;
 
-    SmallVector<Metadata *, 8> MDOps;
-    SmallVector<std::pair<std::string, MDNode *>, 8> NamedMDOps;
-    for (const Attribute &Attribute : F.getAttributes().getFnAttrs()) {
-      if (MDNode *SPIRVMetadata = attributeToDecorateMetadata(Ctx, Attribute))
-        MDOps.push_back(SPIRVMetadata);
-      else if (auto NamedMetadata = attributeToExecModeMetadata(M, Attribute))
-        NamedMDOps.push_back(*NamedMetadata);
+    {
+      // Process all properties on kernels arugments
+      SmallVector<Metadata *, 8> MDOps;
```

Comment:
This will add one a node for each argument, so we may as well reserve the space ahead of time.
```suggestion
      SmallVector<Metadata *, 8> MDOps;
      MDOps.reserve(F.arg_size());
```

Reply:
This is probably not needed since SmallVector already pre-allocated space using the second template arg

Reply:
done